### PR TITLE
RTL3b

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by tcard** [comment]
LGTM

File: ably-ios/ARTTypes.h [hunk 1]
```diff
@@ -47,9 +47,7 @@ typedef NS_ENUM(NSUInteger, ARTRealtimeChannelState) {
     ARTRealtimeChannelInitialised,
     ARTRealtimeChannelAttaching,
     ARTRealtimeChannelAttached,
-    ARTRealtimeChannelDetaching,
```
Comment (mattheworiordan) [inline ably-ios/ARTTypes.h]: State `Detaching` exists, see http://docs.ably.io/client-lib-development-guide/features/#RTL2a, why was this removed?

File: ablySpec/RealtimeClientChannel.swift [hunk 2]
```diff
@@ -13,6 +13,83 @@ class RealtimeClientChannel: QuickSpec {
...
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        let transport = client.transport as! TestProxyTransport
+                        transport.actionsIgnored += [.Attached]
+
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+                        client.close()
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
```
Comment (mattheworiordan) [inline ablySpec/RealtimeClientChannel.swift]: Shouldn't this be eventually as the state after a `close` would be `closing` and the channel state would still be attached at this point?

File: ablySpec/RealtimeClientChannel.swift [hunk 3]
```diff
@@ -13,6 +13,83 @@ class RealtimeClientChannel: QuickSpec {
...
+                        let options = AblyTests.commonAppSetup()
+                        let client = ARTRealtime(options: options)
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+                        client.close()
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
```
Comment (mattheworiordan) [inline ablySpec/RealtimeClientChannel.swift]: As above, I wouldn't expect the channel state to change this quickly

File: ably-ios/ARTTypes.h (hunk 1 above)
Comment (ricardopereira) [inline ably-ios/ARTTypes.h]: I changed it according to http://docs.ably.io/realtime/channels-messages/ _Channel states_ topic. I will undo.

Comment (mattheworiordan) [inline ably-ios/ARTTypes.h]: As mentioned before, the only definitive spec is at http://docs.ably.io/client-lib-development-guide/features/.  I am working on bringing all the other docs up to date, but they're not there yet.

File: ablySpec/RealtimeClientChannel.swift (hunk 2 above)
Comment (ricardopereira) [inline ablySpec/RealtimeClientChannel.swift]: Yes, you're right. #191 is breaking this test.
I will fix the test.

**Comment by ricardopereira** [comment]
Rebased with changes.

**Comment by mattheworiordan** [comment]
LGTM
